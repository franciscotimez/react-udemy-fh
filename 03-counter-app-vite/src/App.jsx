import PropTypes from "prop-types";

const newMessage = 'Francisco';
const obj = { 
  title: 'Hola',
  message: 'Mundo cruel'
}

export const App = ({title, subTitle}) => {

  return (
    <>
      <h1>{ title }</h1>
      <code>{ JSON.stringify(obj) }</code>
      <p>{ subTitle }</p>
    </>
  );
};

App.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number
}

App.defaultProps = {
  title: 'No Hay titulo',
  subTitle: 32
}