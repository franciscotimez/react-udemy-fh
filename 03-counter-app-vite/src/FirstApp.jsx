import PropTypes from "prop-types";

export const FirstApp = ({title, subTitle, name}) => {
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'Francisco Timez',
    subTitle: 'No hay subTitle'
}