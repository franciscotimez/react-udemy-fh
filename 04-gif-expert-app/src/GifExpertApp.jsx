

export const GifExpertApp = () => {
  return (
    <div>GifExpertApp</div>
  )
}
