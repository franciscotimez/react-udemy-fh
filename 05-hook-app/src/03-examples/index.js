
export * from './LoadingQuote'
export * from './Quote'