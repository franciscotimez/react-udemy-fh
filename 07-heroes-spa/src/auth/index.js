
export * from './pages/LoginPage'
export * from './context'
export * from './types/types'