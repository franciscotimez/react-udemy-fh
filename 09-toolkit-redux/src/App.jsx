import reactLogo from './assets/react.svg';
import './App.css';
import { useDispatch, useSelector } from 'react-redux';
import { decrement, increment, incrementBy } from './store/slices/counter/counterSlice';

function App() {
  const { counter } = useSelector(state => state.counter);
  const dispatch = useDispatch();

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src="/vite.svg" className="logo" alt="Vite logo" />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>count is {counter}</h1>
      <div className="card">
        <button onClick={() => { dispatch(increment()); }}>
          Increment
        </button>
        <button onClick={() => { dispatch(decrement()); }}>
          Decrement
        </button>
        <button onClick={() => { dispatch(incrementBy(2)); }}>
          Increment by 2
        </button>
      </div>
    </div>
  );
}

export default App;
