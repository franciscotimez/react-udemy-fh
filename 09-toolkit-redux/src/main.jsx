import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App'
import './index.css'
import { store } from './store';
import { Provider } from 'react-redux';
import { PokemonApp } from './PokemonApp';
import { TodoApp } from './TodoApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Provider store={store}>
      {/* <App /> */}
      {/* <PokemonApp /> */}
      <TodoApp />
    </Provider>
  </React.StrictMode>
)
