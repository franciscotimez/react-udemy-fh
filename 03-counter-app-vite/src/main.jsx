import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './App';
import { CounterApp } from './components/CounterApp';
import { FirstApp } from './FirstApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp initValue={0} />
        {/* <FirstApp title='Hola'/> */}
    </React.StrictMode>
);