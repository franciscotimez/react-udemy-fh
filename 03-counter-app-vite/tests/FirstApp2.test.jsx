import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
  
  const title = "Hola, soy Goku";
  const subTitle = "Hola, soy un subtitulo";

  test('deberia hacer match con el snapshot', () => {

    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();

  });

  test('deberia tener el titulo "Hola, soy Goku"', () => {

    render(<FirstApp title={title} />);
    expect( screen.getByText(title) ).toBeTruthy()
    // screen.debug()

  });

  test('debe tener el titulo en un h1', () => {

    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    );

    expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain(title);
  });

  test('debe de mostrar el subTitle enviado por props', () => {

    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    );

    expect( screen.getByText(subTitle).innerHTML ).toContain(subTitle);
  });
});