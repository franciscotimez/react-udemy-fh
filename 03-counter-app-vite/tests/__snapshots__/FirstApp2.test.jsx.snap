// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> deberia hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    No hay subTitle
  </p>
  <p>
    Francisco Timez
  </p>
</div>
`;
